### ISSUE TITLE
how to test the performence in the test dataset

### ISSUE THREAD

**Issue by namenotexist** [xiangwang1223/neural_graph_collaborative_filtering#23]

**Comment by namenotexist** [comment]
why the training process so slow

**Comment by wanjunhong0** [comment]
In my experience, u need a GPU. Tested on 2080ti, 20s for 1 epoch while without GPU, it took 800s 1 epoch.

**Comment by namenotexist** [comment]
why did i often occur "loss is nan"

**Comment by xiangwang1223** [comment]
You can try the following code to modify the BPR loss, in order to avoid NAN:

loss = tf.reduce_sum(tf.nn.softplus(-(pos_result - neg_result)))

**Comment by namenotexist** [comment]
you mean reduce mean?

**Comment by namenotexist** [comment]
why did i can only use 149M GPU memory, which has a full memory of 10GB

**Comment by xiangwang1223** [comment]
I have no idea... Have you installed the tensorflow-gpu successfully?

**Comment by namenotexist** [comment]
> I have no idea... Have you installed the tensorflow-gpu successfully?
i fix it . because  install tensorflow-gpu 1.14.0,when i change it to tensorflow 1.8.0, it works
